src/modules/apps/logs.ts

```diff
@@ -0,0 +1,79 @@
+import { getManifest } from '../../manifest'
+import userAgent from '../../user-agent'
+import log from '../../logger'
+import { AuthType } from '@vtex/api'
+import { CustomEventSource } from '../../lib/sse/CustomEventSource'
+import { SessionManager } from '../../lib/session/SessionManager'
+import { randomBytes } from 'crypto'
+
+const SKIDDER_MAJOR = 1
+
+export default async (app: string, options) => {
+  const session = SessionManager.getSingleton()
+  const { account, workspace, token, userLogged } = session
+
+  const conf = {
+    headers: {
+      Authorization: `${AuthType.bearer} ${token}`,
+      'user-agent': `${userAgent}${options.past ? `#${randomBytes(8).toString('hex')}` : `#${userLogged}`}`,
+    },
+  }
+
+  try {
+    const manifest = await getManifest()
+    app = app || manifest.name
+  } catch (err) {
+    // manifest file was not found
+
+    if (!account) {
+      console.error('vendor could not be specified, are you logged in?')
+      throw err
+    }
+
+    if (!options.all && !app) {
+      console.error('app could not be specified. Did you forget --all?')
+      throw err
+    }
+  }
+
+  if (options.all) {
+    app = ''
+  }
+
+  let uri = `http://infra.io.vtex.com/skidder/v${SKIDDER_MAJOR}/${account}/${workspace}/logs/stream`
+  if (app) {
+    uri += `/${app}`
+  }
+
+  function createLogEventSource() {
+    const es = new CustomEventSource(uri, conf)
+    const streamLocator = `${account}${app ? `.${app}` : ''}`
+    log.info(`Connecting to logs stream for ${streamLocator}`)
+    log.debug(`Stream URI ${uri}`)
+
+    es.onopen = () => {
+      log.info(`Listening to ${streamLocator}'s logs`)
+    }
+
+    es.onerror = err => {
+      log.error(`Error reading logs: ${err.message}`)
+    }
+
+    es.onclose = () => {
+      log.warn('SSE connection closed. Reconnecting.')
+      createLogEventSource()
+    }
```

Comment:
I think this is not necessary - the underlying eventSource implementation already deals with reconnection, so the `onclose` addition on `CustomEventSource` can be removed as well